tests/client/test_headers.py

```diff
@@ -181,3 +181,22 @@ async def test_host_with_non_default_port_in_url():
             "authorization": "Basic dXNlcm5hbWU6cGFzc3dvcmQ=",
         }
     }
+
+
+@pytest.mark.asyncio
+async def test_none_headers():
+    url = "http://example.org/echo_headers"
+    value: typing.Optional[str] = None
+    headers = {"foo": value}
```

Comment:
(Nit) Does this simplify to…? (Also a suggestion to use a perhaps more explanatory header name, to suggest "this header will be ignored".)

```suggestion
    headers = {"x-ignore": None}
```